src/config/ConfigManager.cpp

```diff
@@ -2,41 +2,43 @@
 #include "../Hyprpaper.hpp"
 
 CConfigManager::CConfigManager() {
-    // init the entire thing
+    // Initialize the configuration
+    // Read file from default location
+    // or from an explicit location given by user
 
     std::string configPath;
     if (g_pHyprpaper->m_szExplicitConfigPath.empty()) {
-        const char *const ENVHOME = getenv("HOME");
-        configPath = ENVHOME + (std::string) "/.config/hypr/hyprpaper.conf";
-    }
-    else {
+        const char* const ENVHOME = getenv("HOME");
+        configPath = ENVHOME + std::string("/.config/hypr/hyprpaper.conf");
+    } else {
         configPath = g_pHyprpaper->m_szExplicitConfigPath;
     }
 
     std::ifstream ifs;
     ifs.open(configPath);
 
     if (!ifs.good()) {
-        Debug::log(CRIT, "Hyprpaper was not provided a config!");
+        if (g_pHyprpaper->m_szExplicitConfigPath.empty()) {
+            Debug::log(CRIT, "No config file provided. Default config file `/.config/hypr/hyprpaper.conf` couldn't be opened.");
+        } else {
+            Debug::log(CRIT, "No config file provided. Specified file `%s` couldn't be opened.", configPath);
+        }
         exit(1);
     }
 
     std::string line = "";
     int linenum = 1;
     if (ifs.is_open()) {
         while (std::getline(ifs, line)) {
-            // Read line by line.
-            try {
-                parseLine(line);
-            } catch (...) {
-                Debug::log(ERR, "Error reading line from config. Line:");
-                Debug::log(NONE, "%s", line.c_str());
-
-                parseError += "Config error at line " + std::to_string(linenum) + ": Line parsing error.";
-            }
+            // Read line by line
+            // The cause of the error is
+            // written in this->parseError
+            parseLine(line);
```

Comment:
dont remove the try block

Reply:
The change has been reverted in the last commit. Try/block is now present.